scripts/library.sh

```diff
@@ -445,7 +445,16 @@ function get_canonical_path() {
 # Return the base url we use to build the actual knative yaml sources.
 function get_knative_base_yaml_source() {
   local knative_base_yaml_source="https://storage.googleapis.com/knative-nightly/@/latest"
-  local branch_name="$(git rev-parse --abbrev-ref HEAD)"
+  local branch_name=""
+  # For non-periodic jobs, we get the branch name from env var set by Prow - https://github.com/kubernetes/test-infra/blob/abcd35c4dbfb0feadd09fc452b533222e3a16b29/prow/jobs.md
+  if [[ -n "${PULL_BASE_REF}" ]]; then
+    branch_name="${PULL_BASE_REF}"
+  fi
+  # For periodic jobs, we only care if it's running against a release branch.
+  # If RELEASE_VERSION is set, branch_name will be the actual release version, like 0.5. Otherwise it will still be an empty string.
+  if [[ -n "${RELEASE_VERSION}" ]]; then
```

Comment:
I like `git rev-parse` better, as `RELEASE_VERSION` doesn't exist when debugging locally

Reply:
Made some changes. Please check the latest code.